core/contracts/Voting.sol

```diff
@@ -113,55 +122,66 @@ contract Voting is Testable {
     function requestPrice(bytes32 identifier, uint time) external returns (uint expectedTime) {
         // TODO: we may want to allow future price requests and/or add a delay so that the price has enough time to be
         // widely distributed and agreed upon before the vote. 
-        require(time < getCurrentTime());
+        uint blockTime = getCurrentTime();
+        require(time < blockTime);
 
+        uint currentRoundId = voteTiming.computeCurrentRoundId(blockTime);
         uint priceResolutionRound = _getPriceResolution(identifier, time).lastVotingRound;
-        uint secondsInWeek = 60*60*24*7;
 
         if (priceResolutionRound == 0) {
             // Price has never been requested.
+
+            // Must ensure the round is updated here so the requested price will be voted on in the next commit cycle.
+            // TODO: this may be an expensive operation - may make sense to have a public updateRound() method to handle
+            // extreme cases.
+            _updateRound(blockTime);
```

Comment:
Maybe worth adding a comment here explaining that we don't update the round unless we have to, because we want to push the burden of doing the update onto voters, not financial contracts (indirectly token sponsors). [is that what we want to do?]

Reply:
Added a comment and moved the call up.